### PR TITLE
[FIXED JENKINS-38534] Isolate the code that requires the `Jenkins` class to be loaded from an agent code path

### PR DESCRIPTION
**Pull request by stephenc**
[JENKINS-38534](https://issues.jenkins-ci.org/browse/JENKINS-38534)

@jenkinsci/code-reviewers @reviewbybees (FYI: @jtnord @andresrc)

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
🐝

**Review by jtnord — APPROVED** [review]
:bee:

**Comment by stephenc** [comment]
@reviewbybees done

**Review by daniel-beck — APPROVED** [review]

**Comment by batmat** [comment]
:bee:

File: core/src/main/java/hudson/util/ProcessKillingVeto.java [hunk 1]
```diff
@@ -74,11 +75,20 @@ public VetoCause(@Nonnull String message) {
...
-            return Collections.emptyList(); // we are remote, no body gets to veto
-        return jenkins.getExtensionList(ProcessKillingVeto.class);
+        if (JenkinsJVM.isJenkinsJVM()) {
+            return _all();
+        }
+        return Collections.emptyList();
+    }
+
+    /**
+     * As classloading is lazy, the classes referenced in this method will not be resolved
```
Comment (jglick) [inline core/src/main/java/hudson/util/ProcessKillingVeto.java]: At least in the Sun/Oracle JVM.

**Review by jglick — APPROVED** [review]
:bee:

**Pull request merged**